ethcore/sync/src/lib.rs

```diff
@@ -21,47 +21,10 @@
 //! https://github.com/ethereum/wiki/wiki/Ethereum-Wire-Protocol
 //!
 
-extern crate client_traits;
-extern crate common_types as types;
-extern crate ethcore_io as io;
-extern crate ethcore_light as light;
-extern crate ethcore_network as network;
-extern crate ethcore_network_devp2p as devp2p;
-extern crate ethcore_private_tx;
-extern crate ethereum_types;
-extern crate ethkey;
-extern crate fastmap;
-extern crate futures;
-extern crate keccak_hash as hash;
-extern crate parity_bytes as bytes;
-extern crate parity_runtime;
-extern crate parking_lot;
-extern crate rand;
-extern crate rlp;
-extern crate snapshot;
-extern crate triehash_ethereum;
-
-#[cfg(test)] extern crate engine;
-#[cfg(test)] extern crate env_logger;
-#[cfg(test)] extern crate ethcore;
-#[cfg(test)] extern crate kvdb_memorydb;
-#[cfg(test)] extern crate machine;
-#[cfg(test)] extern crate rand_xorshift;
-#[cfg(test)] extern crate rustc_hex;
-#[cfg(test)] extern crate spec;
-
-#[macro_use]
-extern crate enum_primitive;
-#[macro_use]
-extern crate macros;
-#[macro_use]
-extern crate log;
-extern crate parity_util_mem;
-#[macro_use]
-extern crate parity_util_mem as malloc_size_of;
-#[macro_use]
-extern crate trace_time;
+// needed to make the procedural macro `MallocSizeOf` to work
+#[macro_use] extern crate parity_util_mem as malloc_size_of;
```

Comment:
@cheme I've run into this as well in other crates. Are we using it wrong or is this a "known issue"?

Reply:
I think the crate need `proc-macro = true` in the manifests to work?